src/Documenter.jl

```diff
@@ -554,6 +554,7 @@ function git_push(
         try
             run(`git checkout -b $branch upstream/$branch`)
         catch e
+            @info "fatal error regarding $branch is expected on first deployment"
```

Comment:
I guess it could be more detailed while we're at it. E.g. something like this?

```suggestion
            @info """
            Checking out $branch failed, creating a new orphaned branch.
            This usually happens when deploying to a repository for the first time and
            the $branch branch does not exist yet. The fatal error above is expected output
            from Git in this situation.
            """
```